src/mlpack/methods/preprocess/load_save_image_main.cpp

```diff
@@ -0,0 +1,145 @@
+PARAM_VECTOR_IN_REQ(string, "input", "Image filenames which have to "
+    "be loaded/saved.", "i");
+
+PARAM_INT_IN("width", "Width of the image", "W", 256);
+PARAM_INT_IN("channel", "Number of channel", "C",  3);
+
+PARAM_MATRIX_OUT("output", "Matrix to save images data to.", "o");
+
+PARAM_INT_IN("quality", "Compression of the image if saved as jpg (0-100).",
+    "q", 90);
+
+PARAM_FLAG("transpose", "Loaded dataset to be transposed", "t");
+
+PARAM_INT_IN("height", "Height of the images", "H", 256);
+PARAM_FLAG("save", "Save a dataset as images", "s");
+PARAM_MATRIX_IN("dataset", "Input matrix to save as images.", "I");
+
+// Loading/saving of a Image Info model.
+PARAM_MODEL_IN(ImageInfo, "input_model", "Input Image Info model.", "m");
+PARAM_MODEL_OUT(ImageInfo, "output_model", "Output Image Info model.", "M");
+
+#ifdef HAS_STB // Compile this only if stb is present.
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  data::ImageInfo* info;
+
+  Timer::Start("Loading/Saving Image");
+  if (CLI::HasParam("input_model"))
+  {
+    info = CLI::GetParam<ImageInfo*>("input_model");
+  }
+  else
+  {
+    if (!CLI::HasParam("width") || !CLI::HasParam("height") ||
+        !CLI::HasParam("channel"))
+    {
+      throw std::runtime_error("Please provide height, width and "
+          "number of channels of the images.");
+    }
+    // Positive value for width.
+    RequireParamValue<int>("width", [](int x) { return x >= 0;}, true,
+        "width must be positive");
+    // Positive value for height.
+    RequireParamValue<int>("height", [](int x) { return x >= 0;}, true,
+        "height must be positive");
+    // Positive value for channel.
+    RequireParamValue<int>("channel", [](int x) { return x >= 0;}, true,
+        "channel must be positive");
+    // Positive value for quality.
+    RequireParamValue<int>("quality", [](int x) { return x >= 0;}, true,
+        "quality must be positive");
+
+    const size_t& height = CLI::GetParam<int>("height");
+    const size_t& width = CLI::GetParam<int>("width");
+    const size_t& channel = CLI::GetParam<int>("channel");
+    const size_t& quality = CLI::GetParam<int>("quality");
+    info = new data::ImageInfo(width, height, channel, quality);
+  }
+  const vector<string> fileNames =
+      CLI::GetParam<vector<string> >("input");
+  arma::mat out;
+  if (!CLI::HasParam("save"))
+  {
+    Load(fileNames, out, *info, false, !CLI::HasParam("transpose"));
+    if (CLI::HasParam("output"))
+      CLI::GetParam<arma::mat>("output") = std::move(out);
+  }
+  if (CLI::HasParam("save"))
+  {
+    if (!CLI::HasParam("dataset"))
+    {
+      throw std::runtime_error("Please provide a input matrix to save "
+          "images from");
+    }
+    Save(fileNames, CLI::GetParam<arma::mat> ("dataset"), *info, false,
+        !CLI::HasParam("transpose"));
+  }
+  if (CLI::HasParam("output_model"))
+    CLI::GetParam<ImageInfo*> ("output_model") = info;
+}
+#else
+
+static void mlpackMain() {}
```

Comment:
It might be useful to put an `#error` in here, since if `HAS_STB` isn't defined this program isn't going to do anything useful anyway.  Honestly if CMake only builds this when STB is found, then I don't see the need for the `#if HAS_STB` in this program---it's a redundant check.

Reply:
Removed